boards/DCM/Src/App/torquevectoring/App_TorqueVectoring.c

```diff
@@ -0,0 +1,97 @@
+
+    PID pid_traction_control;
+    App_PID_Init(&pid_traction_control, &PID_TRACTION_CONTROL_CONFIG);
+    traction_control_inputs.pid = &pid_traction_control;
+
+    App_Timer_InitTimer(&pid_timeout, PID_TIMEOUT_ms);
+}
+
+void App_TorqueVectoring_Run(void)
+{
+    // FIXME: Replace with CAN messages
+    float accelerator_pedal_percent = 2.0;
+    bool regen_engaged = true;
+
+    if (accelerator_pedal_percent > 0.0f) {
+        // Reset control loops if timeout elapsed
+        TimerState timeout = App_Timer_UpdateAndGetState(&pid_timeout);
+        if (timeout == TIMER_STATE_EXPIRED) {
+            App_PID_RequestReset(&pid_power_correction);
+            pid_power_correction_factor = 1.0f;
+            App_PID_RequestReset(traction_control_inputs.pid);
+
+        }
+        App_Timer_Restart(&pid_timeout);
+
+        // Get CAN messages re-used by multiple modules
+        float motor_speed_left_rpm = (float) App_CanRx_INVL_MotorPositionInfo_MotorSpeed_Get();
+        float motor_speed_right_rpm = (float) App_CanRx_INVR_MotorPositionInfo_MotorSpeed_Get();
+
+        // Power Limiting
+        power_limiting_inputs.left_motor_temp_C = App_CanRx_INVL_Temperatures3_MotorTemperature_Get();
+        power_limiting_inputs.right_motor_temp_C = App_CanRx_INVR_Temperatures3_MotorTemperature_Get();
+        // TODO: Available power will at some point be replaced by current + voltage messages
+        power_limiting_inputs.available_battery_power_kW = App_CanRx_BMS_AvailablePower_AvailablePower_Get();
+        //power_limiting_inputs.accelerator_pedal_percent = App_CanRx_
+        float estimated_power_limit = App_PowerLimiting_ComputeMaxPower(&power_limiting_inputs);
+
+        // Power limit correction
+        float power_limit = estimated_power_limit * pid_power_correction_factor;
+
+        // Active Differential
+        active_differential_inputs.power_max_kW = power_limit;
+        active_differential_inputs.motor_speed_left_rpm = motor_speed_left_rpm;
+        active_differential_inputs.motor_speed_right_rpm = motor_speed_right_rpm;
+        //active_differential_inputs.wheel_angle_deg = App_Can
+        App_ActiveDifferential_ComputeTorque(&active_differential_inputs, &active_differential_outputs);
+
+        // Traction Control
+        traction_control_inputs.motor_speed_left_rpm = motor_speed_left_rpm;
+        traction_control_inputs.motor_speed_right_rpm = motor_speed_right_rpm;
+        traction_control_inputs.torque_left_Nm = active_differential_outputs.torque_left_Nm;
+        traction_control_inputs.torque_right_Nm = active_differential_outputs.torque_right_Nm;
+        App_TractionControl_ComputeTorque(&traction_control_inputs, &traction_control_outputs);
+
+        // Inverter Torque Requests
+        App_CanTx_DCM_LeftInverterCommand_TorqueCommand_Set(traction_control_outputs.torque_left_final_Nm);
+        App_CanTx_DCM_RightInverterCommand_TorqueCommand_Set(traction_control_outputs.torque_right_final_Nm);
```

Comment:
Final torque clamp? Doesn't look like ComputeTorque does that, and imo it's better to do it here anyway to be super explicit about clamping